tests/auto/rt_auto.py

```diff
@@ -0,0 +1,304 @@
+
+    def add_pr_label(self):
+        ''' adds the pull request label that initiated the job run to PR'''
+        self.logger.info(f'Adding Label: {self.preq_dict["label"]}')
+        self.preq_dict['preq'].add_to_labels(self.preq_dict['label'])
+
+    def send_log_as_comment(self):
+        with open('rt_auto.log', 'r') as f:
+            filedata = f.read()
+        self.preq_dict['preq'].create_issue_comment(filedata)
+
+    def run_commands(self, commands_with_cwd):
+        logger = logging.getLogger('RUN_COMMANDS')
+        for command, in_cwd in commands_with_cwd:
+            logger.info(f'Running "{command}" in location "{in_cwd}"')
+            try:
+                output = subprocess.Popen(command, shell=True, cwd=in_cwd, stdout=subprocess.PIPE, stderr=subprocess.STDOUT)
+                out, err = output.communicate()
+                out = [] if not out else out.decode('utf8').split('\n')
+                err = [] if not err else err.decode('utf8').split('\n')
+            except Exception as e:
+                self.add_pr_label()
+                logger.critical(e)
+                [logger.critical(f'stdout: {item}') for item in out if not None]
+                [logger.critical(f'stderr: {eitem}') for eitem in err if not None]
+                assert(e)
+            else:
+                logger.info(f'Finished running: {command}')
+                [logger.debug(f'stdout: {item}') for item in out if not None]
+                [logger.debug(f'stderr: {eitem}') for eitem in err if not None]
+
+    def remove_pr_dir(self):
+        logger = logging.getLogger('JOB/REMOVE_PR_DIR')
+        pr_dir_str = f'{self.machine["workdir"]}/{str(self.preq_dict["preq"].id)}'
+        rm_command = [f'rm -rf {pr_dir_str}', os.getcwd()]
+        logger.info(f'Running "{rm_command}"')
+        self.run_commands(rm_command)
+
+    def clone_pr_repo(self):
+        ''' clone the GitHub pull request repo, via command line '''
+        logger = logging.getLogger('JOB/CLONE_PR_REPO')
+        repo_name = self.preq_dict['preq'].head.repo.name
+        self.branch = self.preq_dict['preq'].head.ref
+        git_url = self.preq_dict['preq'].head.repo.html_url.split('//')
+        git_url = f'{git_url[0]}//${{ghapitoken}}@{git_url[1]}'
+        logger.info(f'GIT URL: {git_url}')
+        logger.info('Starting repo clone')
+        repo_dir_str = f'{self.machine["workdir"]}/{str(self.preq_dict["preq"].id)}/{datetime.datetime.now().strftime("%Y%m%d%H%M%S")}'
+
+        create_repo_commands = [
+            [f'mkdir -p "{repo_dir_str}"', self.machine['workdir']],
+            [f'git clone -b {self.branch} {git_url}', repo_dir_str],
+            [f'git submodule update --init --recursive', f'{repo_dir_str}/{repo_name}']
+        ]
+
+        self.run_commands(create_repo_commands)
+
+        logger.info('Finished repo clone')
+        self.pr_repo_loc = repo_dir_str+"/"+repo_name
+        return self.pr_repo_loc
+
+    def runFunction(self):
```

Comment:
Consider renaming this function to `run_function`, to be consistent with the naming scheme of other functions.

Reply:
done, committing shortly